VM/include/lua.h

```diff
@@ -356,6 +356,8 @@ struct lua_Callbacks
     void (*debugstep)(lua_State* L, lua_Debug* ar);      /* gets called after each instruction in single step mode */
     void (*debuginterrupt)(lua_State* L, lua_Debug* ar); /* gets called when thread execution is interrupted by break in another thread */
     void (*debugprotectederror)(lua_State* L);           /* gets called when protected call results in an error */
+
+    void* userdata; /* arbitrary userdata pointer that is never overwritten by Luau */
```

Comment:
> I'd rather break ABI right now while we're still super early
I actually think you shouldn't really worry about ABI. Applications that want to use Luau to its full potential should be statically compiling a pinned version into their executable. That way they can access implementation details without fear of updates breaking compatibility, because an update would just mean recompiling their application.